### PR TITLE
“add refill_unblock_check” spec update, part II

### PR DESCRIPTION
**Pull request by mktnk3**
This PR completes the `refill_unblock_check` spec change for `sched_context_bind_tcb`.

### CODE REVIEW

**Review by lsf37 — APPROVED** [review]
Looks good from my side!

File: proof/refine/ARM/Tcb_R.thy [hunk 1]
```diff
@@ -2049,7 +2066,8 @@ lemma schedContextBindTCB_corres:
     apply (rule corres_symb_exec_r')
        apply (rule corres_split_nor)
           apply (rule corres_split_nor)
-             apply (rule corres_split_nor)
+             apply (rule corres_split[OF ifCondRefillUnblockCheck_corres])
+          apply (rule corres_split_nor)
```
Comment (mktnk3) [inline proof/refine/ARM/Tcb_R.thy:2070]: fix indentation here

File: proof/invariant-abstract/DetSchedSchedule_AI.thy [hunk 2]
```diff
@@ -22631,17 +22730,27 @@ lemma sched_context_bind_tcb_cur_sc_in_release_q_imp_zero_consumed[wp]:
...
   apply (rule hoare_seq_ext[rotated])
    apply (rule sched_context_bind_tcb_cur_sc_in_release_q_imp_zero_consumed_helper)
-
-  apply (wpsimp wp: sched_context_resume_cur_sc_in_release_q_imp_zero_consumed)
-  apply (clarsimp simp: heap_refs_inv_def split: if_splits)
-  apply (rule cur_sc_more_than_ready_imp_cur_sc_ready_and_sufficient; blast?)
+  apply (clarsimp simp: maybeM_def bind_assoc)
+  apply (rule hoare_seq_ext[OF _ get_sched_context_sp])
+  apply (rule hoare_seq_ext[OF _ gets_sp])
+  apply (rule_tac P="K (sc_ptr = xa)" in hoare_pre_tautI; simp)
```
Comment (michaelmcinerney) [inline proof/invariant-abstract/DetSchedSchedule_AI.thy:22743]: `rename_tac` for this `xa`, and there are two `apply`s on the line before `done`

File: proof/refine/ARM/Tcb_R.thy [hunk 3]
```diff
@@ -2033,13 +2034,29 @@ lemma sched_context_resume_weak_valid_sched_action:
   unfolding sched_context_resume_def
   by (wpsimp wp: postpone_weak_valid_sched_action thread_get_wp' is_schedulable_wp')
 
+lemma refill_unblock_check_weak_valid_sched_action[wp]:
+  "\<lbrace>weak_valid_sched_action and active_sc_valid_refills\<rbrace>
+    refill_unblock_check sc_ptr
```
Comment (michaelmcinerney) [inline proof/refine/ARM/Tcb_R.thy:2039]: Indentation

**Review by michaelmcinerney — APPROVED** [review]
Just a couple nitpicks

**Review by corlewis — APPROVED** [review]

**Pull request merged**